File: scrimage-core/src/main/scala/com/sksamuel/scrimage/nio/ImageWriter.scala
package com.sksamuel.scrimage.nio

import java.io.{IOException, OutputStream}

import com.sksamuel.scrimage.Image

/**
  * Typeclass supporting writing of an Image to an array of bytes in a specified format, eg JPEG
  */
trait ImageWriter {
  @throws[IOException]
  def write(image: Image, out: OutputStream): Unit
}

object ImageWriter {
  implicit def default: PngWriter = PngWriter.MaxCompression
}

File: scrimage-core/src/main/java/com/sksamuel/scrimage/nio/ImageWriter.java
package com.sksamuel.scrimage.nio;

import com.sksamuel.scrimage.AwtImage;
import com.sksamuel.scrimage.metadata.ImageMetadata;

import java.io.IOException;
import java.io.OutputStream;

/**
 * Interface supporting writing of an Image to an array of bytes in a specified format, eg JPEG
 *
 * @see com.sksamuel.scrimage.nio.JpegWriter
 * @see com.sksamuel.scrimage.nio.PngWriter
 * @see com.sksamuel.scrimage.nio.GifWriter
 */
public interface ImageWriter {
   void write(AwtImage image, ImageMetadata metadata, OutputStream out) throws IOException;
}
